app/code/Magento/MediaStorage/Model/File/Validator/NotProtectedExtension.php

```diff
@@ -78,6 +78,9 @@ protected function _initProtectedFileExtensions()
     {
         if (!$this->_protectedFileExtensions) {
             $extensions = $this->getProtectedFileExtensions();
+            if ($extensions === null) {
+                $extensions = [];
+            }
             if (is_string($extensions)) {
```

Comment:
if(is_null($extension)) is not allowed. Then why can i see _is_string($extension)_ over here ? 
Are there special rules for Magento Core developers ?

Reply:
> why can i see is_string($extension) over here ?

I don't get your logic. What would be a replacement for `is_string`?

Reply:
Then why change is_null then ? Thus is_string is only good because you can't change it. Also the code is not cleaner. You should try to code cleanly, not like your sniffer wants to.